From a9590fe148c03cb4157b56255357419cb4e14124 Mon Sep 17 00:00:00 2001
From: weiping <zhangweiping@didichuxing.com>
Date: Thu, 22 Jun 2017 23:06:56 +0800
Subject: [PATCH] blk-mq: remove double set queue_num
Git-commit: a9590fe148c03cb4157b56255357419cb4e14124
Patch-mainline: v4.13-rc1
References: fate#322738,fate#322919,fate#322950,fate#323773

hwctx's queue_num has been set prior call blk_mq_init_hctx, so no need
set it again.

Signed-off-by: weiping <zhangweiping@didichuxing.com>
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Signed-off-by: Coly Li <colyli@suse.de>

---
 block/blk-mq.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 2caac30e128a..c56b64ae1741 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1910,7 +1910,6 @@ static int blk_mq_init_hctx(struct request_queue *q,
 	spin_lock_init(&hctx->lock);
 	INIT_LIST_HEAD(&hctx->dispatch);
 	hctx->queue = q;
-	hctx->queue_num = hctx_idx;
 	hctx->flags = set->flags & ~BLK_MQ_F_TAG_SHARED;
 
 	cpuhp_state_add_instance_nocalls(CPUHP_BLK_MQ_DEAD, &hctx->cpuhp_dead);
-- 
2.13.1

